farmhelper-mod/src/main/java/com/jelly/farmhelper/features/PetSwapper.java

```diff
@@ -0,0 +1,220 @@
+package com.jelly.farmhelper.features;
+
+import com.jelly.farmhelper.config.Config;
+import com.jelly.farmhelper.macros.MacroHandler;
+import com.jelly.farmhelper.utils.Clock;
+import com.jelly.farmhelper.utils.LogUtils;
+import com.jelly.farmhelper.utils.PlayerUtils;
+import net.minecraft.client.Minecraft;
+import net.minecraft.client.gui.inventory.GuiChest;
+import net.minecraft.inventory.ContainerChest;
+import net.minecraft.item.ItemStack;
+import net.minecraft.util.StringUtils;
+import net.minecraftforge.fml.common.eventhandler.SubscribeEvent;
+import net.minecraftforge.fml.common.gameevent.TickEvent;
+
+public class PetSwapper {
+
+    private static Minecraft mc = Minecraft.getMinecraft();
+    public String swappedPet;
+    private String originalPet;
+    private static Clock delay = new Clock();
+
+    //credit to Nirox for the clicktypes
+    public enum ClickType {
+        PICKUP,
+        QUICK_MOVE,
+        SWAP,
+        CLONE,
+        THROW,
+        QUICK_CRAFT
+    }
+
+    public enum StatePet {
+        OPEN_MENU,
+        FIND_PET,
+        TURN_PAGE,
+        CLICK_PET,
+        CLOSE_MENU
+
+    }
+
+    private static StatePet state = StatePet.OPEN_MENU;
+    private static ContainerChest chest;
+    private static int petSlot;
+    private static int pageSlot;
+    private static boolean shouldSwap = false;
+
+    public static void swapPets() {
+try {
+        shouldSwap = true;
+        switch (state) {
+            case OPEN_MENU:
+               MacroHandler.disableCurrentMacro(true);
+                mc.thePlayer.sendChatMessage("/pets");
+                state = StatePet.FIND_PET;
```

Comment:
Firstly find the current pet and store it in a variable. Clear the variable after turning off the macro and only search for the current pet if the variable is null. You can store index, name, whatever can help you switch back to it. Similar to (or even the same as) the code you've got commented